infrastructure/storage/tests/lmdb.rs

```diff
@@ -330,3 +330,47 @@ fn lmbd_resize_on_create() {
     }
     clean_up(&db_env_name); // In Windows file handles must be released before files can be deleted
 }
+
+#[test]
+fn test_lmdb_resize_before_full() {
+    let db_env_name = "resize_dynamic";
+    {
+        let path = get_path(&db_env_name);
+        std::fs::create_dir(&path).unwrap_or_default();
+        let db_name = "test_full";
+        {
+            // Create db with 1MB capacity
+            let store = LMDBBuilder::new()
+                .set_path(&path)
+                .set_environment_size(1)
+                .set_max_number_of_databases(1)
+                .add_database(&db_name, db::CREATE)
+                .build()
+                .unwrap();
+            let db = store.get_handle(&db_name).unwrap();
+
+            // Add enough data to exceed our 1MB db
+            let value = load_users();
+            // one insertion requires approx 92KB so after ~11 insertions
+            // our 1MB env size should be out of space
+            // however the db should now be allocating additional space as it fills up
+            for key in 0..32 {
+                if let Err(_e) = db.insert(&key, &value) {
+                    // println!("LMDBError {:#?}", _e);
+                    panic!("Failed to resize the LMDB store.");
+                }
+            }
+            let env_info = store.env().info().unwrap();
+            let psize = store.env().stat().unwrap().psize as usize;
+            let page_size_total = psize * env_info.last_pgno;
+            let percent_left = 1.0 - page_size_total as f64 / env_info.mapsize as f64;
+
+            // check the allocated size is now greater than it was initially
+            assert!(page_size_total > 1024 * 1024);
+            assert!(percent_left > 0.0);
```

Comment:
I think in this instance we should also assert that `space left <= (1 + RESIZE_THRESHOLD ) * initial capacity`, to ensure it does not grow out of bounds.

Reply:
Thanks I'll take a look at that

Reply:
I will post new suggestions; ran a 'sync from scratch' test and the resize logic has some problem.